src/scrollspy/scrollspy.ts

```diff
@@ -0,0 +1,51 @@
+import {Directive, ElementRef, Injectable, Input, OnDestroy, OnInit, Renderer} from '@angular/core';
+
+@Directive({selector: '[ngbScrollSpy]'})
+export class NgbScrollSpy implements OnDestroy, OnInit {
+  active: boolean;
+  selector: string;
+
+  constructor(public elementRef: ElementRef, private scrollSpier: NgbScrollSpier) {}
+
+  @Input()
+  set ngbScrollSpy(selector: string) {
+    this.selector = selector;
+  }
+
+  get ngbScrollSpy() { return this.selector; }
+
+  ngOnDestroy(): void { this.scrollSpier.deregister(this); }
+
+  ngOnInit(): void { this.scrollSpier.register(this); }
+}
+
+@Directive({selector: '[ngbScrollTarget]'})
+export class NgbScrollTarget implements OnInit {
+  constructor(private elementRef: ElementRef, private renderer: Renderer, private scrollSpier: NgbScrollSpier) {}
+
+  checkSpies(evt: Event) { this.scrollSpier.spies.forEach(spy => this.checkSpy(spy, evt)); }
+
+  checkSpy(spy: NgbScrollSpy, evt: Event) {
+    let element = this.elementRef.nativeElement;
+    let target = element.querySelector(spy.selector);
+
+    if (!target) {
+      return;
+    }
+
+    this.renderer.setElementClass(
```

Comment:
Shouldn't we have a method like 'setActive(...)' on the `NgbScrollSpy` instead? Just an idea...